apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/ProjectMetricsResourceTest.java

```diff
@@ -530,6 +581,43 @@ private <T extends Number> ProjectMetricResponse<T> getProjectMetrics(
         }
     }
 
+    private <T extends Number> void getMetricsAndAssert(
+            UUID projectId, ProjectMetricRequest request, Instant marker, List<String> names, Class<T> aClass,
+            Map<String, T> minus3, Map<String, T> minus1, Map<String, T> current) {
+        var response = getProjectMetrics(projectId, request, aClass);
+
+        var expected = createExpected(marker, request.interval(), names, minus3, minus1, current);
+
+        // assertions
+        assertThat(response.projectId()).isEqualTo(projectId);
+        assertThat(response.metricType()).isEqualTo(request.metricType());
+        assertThat(response.interval()).isEqualTo(request.interval());
+        assertThat(response.results()).hasSize(names.size());
+
+        assertThat(response.results()).hasSize(expected.size());
+        assertThat(response.results())
+                .usingRecursiveComparison()
+                .withComparatorForType(this::bigDecimalInDelta, BigDecimal.class)
+                .ignoringCollectionOrder()
+                .isEqualTo(expected);
+    }
+
+    private int bigDecimalInDelta(BigDecimal number, BigDecimal number2) {
+        if (number == null) {
+            return number2 == null ? 0 : 1;
+        }
+        if (number2 == null) {
+            return -1;
+        }
+
+        var delta = new BigDecimal(".000001");
```

Comment:
Do we need this delta part? `BigDecimal ` is supposed to be exact without rounding.

Reply:
See findAllAndCalculateFeedbackAvg. We have to do something similar

Reply:
If we are using `BigDecimal` in Java code, then maybe in order to be consistent we could use same types in DB? Like Decimal32, Decimal64, etc.?

Or if precision is not exact anyway due to DB constraints, then why do we use `BigDecimal` in the code? Could be just float/double?

Reply:
Reused existing comparator in https://github.com/comet-ml/opik/pull/728/commits/275c0d3fa558875650b83cc8ed6ba944b33ca08c